hal/armv7m/interrupts.c

```diff
@@ -52,24 +52,40 @@ static void interrupts_nvicSetPriority(s8 irqn, u32 priority)
 }
 
 
-__attribute__((section(".noxip"))) void hal_interruptsEnable(void)
+__attribute__((section(".noxip"))) void hal_interruptsEnableAll(void)
 {
 	__asm__ volatile("cpsie if");
 }
 
 
-__attribute__((section(".noxip"))) void hal_interruptsDisable(void)
+__attribute__((section(".noxip"))) void hal_interruptsDisableAll(void)
 {
 	__asm__ volatile("cpsid if");
 }
 
 
+__attribute__((section(".noxip"))) void hal_interruptsEnable(unsigned int irqn)
+{
+	if (irqn >= 0x10) {
+		interrupts_nvicSetIRQ(irqn - 0x10, 1);
+	}
```

Comment:
You need to handle irq < 16 here too, otherwise disable on SYSTICK won't work (it's an exception with irq number 15)

Reply:
Or just disable all interrupts in case of armv7m timers

Reply:
I have reverted the change for `SYSTICK` in `stm32/l4/timer.c`.